### PR TITLE
Migrate to docusaurus

### PR DESCRIPTION
**Pull request by sniok**

### CODE REVIEW

File: blog/2022-03-16-btfgen.md [hunk 1]
```diff
@@ -237,42 +236,42 @@ to modify the opensnoop, execsnoop and bindsnoop tools.
...
 @@ -187,7 +187,8 @@ int main(int argc, char **argv)
  	libbpf_set_strict_mode(LIBBPF_STRICT_ALL);
  	libbpf_set_print(libbpf_print_fn);
 
--	obj = bindsnoop_bpf__open();
-+	LIBBPF_OPTS(bpf_object_open_opts, opts, .btf_custom_path = "/tmp/vmlinux.btf");
-+	obj = bindsnoop_bpf__open_opts(&opts);
+-	obj: bindsnoop_bpf__open();
++	LIBBPF_OPTS(bpf_object_open_opts, opts, .btf_custom_path: "/tmp/vmlinux.btf");
++	obj: bindsnoop_bpf__open_opts(&opts);
```
[the remark this answers is not in the document]
Comment (sniok) [reply blog/2022-03-16-btfgen.md:246]: yeah. good catch, thanks!

File: docusaurus.config.ts [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+              label: "latest",
+              path: "latest",
+            },
+            [latestVersionName]: {
+              banner: "none",
+            },
+          },
+          sidebarPath: "./sidebars.ts",
+          editUrl:
+            "https://github.com/headlamp-k8s/headlamp-website/edit/main/",
```
Comment (mauriciovasquezbernal) [inline docusaurus.config.ts:67]:
```suggestion
            "https://github.com/inspektor-gadget/inspektor-gadget/edit/main/",
```

File: tools/docs-fetcher.py [hunk 3]
```diff
@@ -60,21 +66,174 @@ def fetch_docs(file_path):
...
+        # Main index page should be first in the sidebar 
+        # use regex to replace weight: anynumber to sidebar_position: 1
+        contents = re.sub(r'weight: \d+', 'sidebar_position: 1', contents)
+
+        # Write the file back
+        with open(file_path, 'w') as f:
+            f.write(contents)
+         
+
+def convert_hugo_to_docusaurus_format(dst_dir, verion_name):
```
Comment (mauriciovasquezbernal) [inline tools/docs-fetcher.py:192]:
```suggestion
def convert_hugo_to_docusaurus_format(dst_dir, version_name):
```

**Review by mauriciovasquezbernal — APPROVED** [review]
A couple of minor comments, already LGTM. 

I tested it with https://github.com/inspektor-gadget/inspektor-gadget/pull/3249 and it worked perfectly.

**Comment by joaquimrocha** [comment]
The blog adds the posts with YYYY/MM/DD/ but in Hugo we had them only with YYYY/MM. I did changes to the Headlamp one so we have this change in order to maintain the links that search engines already indexed. We should do this change here too.

**Comment by sniok** [comment]
updated all blog posts to have a slug that follows the /yyyy/mm/title convention

**Comment by blanquicet** [comment]
Thanks for working on this. I tested it and these are the commands I had to run be able to test it locally:

```bash
$ npm install docusaurus
$ make docs
$ npm run dev
```

The fist two are not documentation, I suggest adding them. 

Additionally, I had some problems because my node version wasn't > 18.0. So I suggest to specify it in the requirements section you added in the README. Same for other dependencies, if applicable. Thanks!

**Comment by burak-ok** [comment]
Looks really good, thanks :)
Since I am a dark theme user I found the following issue: the links have the same color as normal text, making it all white. Same with the Sidebar on the `Docs` page

**Pull request merged**